registry/server/api_topics.go

```diff
@@ -344,13 +345,20 @@ func (s *Server) fetchTopicSet(req *pb.TopicRequest) (TopicSet, error) {
 
 	// Populate all topics.
 	for _, t := range topics {
-		s, _ := s.ZK.GetTopicState(t)
+		st, _ := s.ZK.GetTopicState(t)
+		c, err := s.ZK.GetTopicConfig(t)
+		if err != nil {
```

Comment:
We should probably return an error here + check/return the error in GetTopicState as well. It should be relatively rare that we hit an error here short of connectivity errors to ZK, which at that point the whole request will (should) likely just fail.

Reply:
My logic behind the error check here was in case there was some corruption in zk (delete /configs/topics/topic-name) and if you return an error there, then no topic will be listed even though only one is corrupted. Agree though that hitting an error here is relatively rare.